near-sdk-macros/src/core_impl/abi/abi_generator.rs

```diff
@@ -126,6 +127,34 @@ impl ImplItemMethodInfo {
                         None
                     }
                 }
+                ReturnType::Type(_, ty) if utils::type_is_result(ty) && is_handles_result => {
+                    let ty = if let Some(ty) = utils::extract_ok_type(ty) {
+                        ty
+                    } else {
+                        return syn::Error::new_spanned(
+                            ty,
+                            "Function marked with #[handle_result] should have return type Result<T, PromiseError>",
```

Comment:
Shouldn't necessarily be `PromiseError` for the `Result` error variant right? 
```suggestion
                            "Function marked with #[handle_result] should have return type Result<T, E>",
```

Reply:
Yes, that's correct, only `callback_result` uses that type. For the function error type it can be anything that implements the `FunctionError` trait. Maybe that can be expressed in this comment also

Reply:
Yeah just forgot to update the text after copy-pasting this from somewhere else. Added a comment about E having to implement `FunctionError`, seems like a useful clarification indeed